src/ducks/categories/CategoriesPage.jsx

```diff
@@ -248,43 +254,94 @@ const enhance = Component => props => {
+    groups,
+    accounts,
+    filteringDoc,
+    dateAttribute: 'date'
+  })
+  const initialConnByApplicationDate = makeFilteredTransactionsConn({
     groups,
     accounts,
-    filteringDoc
+    filteringDoc,
+    dateAttribute: 'applicationDate'
   })
-  const conn = useMemo(() => {
+
+  const connByDate = useMemo(() => {
+    return period
+      ? setAutoUpdate(
+          addPeriodToConn({
+            baseConn: initialConnByDate,
+            period,
+            dateAttribute: 'date'
+          })
+        )
+      : setAutoUpdate(initialConnByDate)
+  }, [initialConnByDate, period])
+
+  const connByApplicationDate = useMemo(() => {
     return period
-      ? setAutoUpdate(addPeriodToConn(initialConn, period))
-      : setAutoUpdate(initialConn)
-  }, [initialConn, period])
-  const transactions = useFullyLoadedQuery(conn.query, conn)
+      ? setAutoUpdate(
+          addPeriodToConn({
+            baseConn: initialConnByApplicationDate,
+            period,
+            dateAttribute: 'applicationDate'
+          })
+        )
+      : setAutoUpdate(initialConnByApplicationDate)
+  }, [initialConnByApplicationDate, period])
+
+  const transactionsByDate = useFullyLoadedQuery(connByDate.query, connByDate)
+  const transactionsByApplicationDate = useFullyLoadedQuery(
+    connByApplicationDate.query,
+    connByApplicationDate
+  )
 
   // This is used for loaded transactions to stay rendered while
   // next/previous month transactions are loaded
-  const col = useLast(transactions, (last, cur) => {
+  const colByDate = useLast(transactionsByDate, (last, cur) => {
     return !last || (cur.lastUpdate && !cur.hasMore)
   })
+  const colByApplicationDate = useLast(
+    transactionsByApplicationDate,
+    (last, cur) => {
+      return !last || (cur.lastUpdate && !cur.hasMore)
+    }
+  )
+
+  const transactionsData = useMemo(
+    () =>
+      computeTransactionsByDateAndApplicationDate({
+        transactionsByDate: transactionsByDate.data,
+        transactionsByApplicationDate: transactionsByApplicationDate.data
+      }),
+    [transactionsByApplicationDate.data, transactionsByDate.data]
+  )
 
   const categories = useMemo(() => {
-    return computeCategoriesData(col.data || [])
-  }, [col.data])
+    return computeCategoriesData(transactionsData || [])
+  }, [transactionsData])
+
   return (
     <Component
       {...props}
       accounts={accounts}
       groups={groups}
       settings={settings}
       categories={categories}
-      transactions={col}
+      transactions={colByDate}
+      transactionsByApplicationDate={colByApplicationDate}
       filteringDoc={filteringDoc}
       period={period}
       client={client}
       params={params}
-      filteredTransactionsByAccount={col.data}
+      filteredTransactionsByAccount={transactionsData}
       dispatch={dispatch}
       isFetching={
-        transactions.fetchStatus === 'loading' || transactions.hasMore
+        transactionsByDate.fetchStatus === 'loading' ||
```

Comment:
this boolean value is a bit complex to understand, maybe it should be exported in a hook or some function with some comment explanation, maybe with a small unit test too.

Reply:
We should not rely on === loading. It can be pending. You should use isQueryLoading()